### DISCUSSION TITLE
Pass on click envet in the onclick function of menu table render

### DISCUSSION THREAD

**Discussion by lafray** [vtex/shoreline#442] [General]
Atualemente, se eu faço um menu, o onClick do MenuItem repassa o onClickEvent. Isso possibilita que se o evento for por exemplo um link para outra página, eu possa abrir numa aba separada ou não dependendo das informação no evento. Porém o onClick que o render menu recebe não passa o evento para a função, impossibilitando esse tipo de funcionalidade. Seria um problema vcs repassarem esse objeto do evento sendo que ele já existe no component puro?

**Comment by lafray** [comment]
Atualmente usamos um render custom para o menu (na época não existia um do admin) e não vamos migrar por conta dessa perda de funcionalidade.

O código do nosso custom é o seguinte: 
```
import React, { memo } from 'react'
import type { Seller } from 'vtex.sellers-graphql'
import {
  Box,
  Skeleton,
  Menu,
  MenuButton,
  MenuItem,
  useMenuState,
  IconStorefront,
  IconShoppingCartSimple,
} from '@vtex/admin-ui'
import { defineMessages, useIntl } from 'react-intl'

import type { OnClickEvent } from '../../utils/navigation'
import { goToSellerDetails, goToOrders } from '../../utils/navigation'
import type { ResolverRenderProps } from '../../../typings/TableRander'

const msgPrefix = 'admin/seller-management.sellers-list.table.more'

const messages = defineMessages({
  details: { id: `${msgPrefix}.details` },
  orders: { id: `${msgPrefix}.orders` },
  salesPerformance: { id: `${msgPrefix}.sales-performance` },
  manuLabel: { id: `${msgPrefix}.label` },
})

const MoreMenu = (props: { seller: Seller }) => {
  const { seller } = props
  const { formatMessage } = useIntl()

  const menu = useMenuState({
    orientation: 'vertical',
    placement: 'bottom-start',
  })

  return (
    <Box onClick={(e: React.MouseEvent<HTMLElement>) => e.stopPropagation()}>
      <MenuButton
        state={menu}
        variant="neutralTertiary"
        labelHidden
        aria-label={formatMessage(messages.manuLabel)}
      />
      <Menu aria-label={formatMessage(messages.manuLabel)} state={menu}>
        <MenuItem
          icon={<IconStorefront />}
          onClick={(e: OnClickEvent) => goToSellerDetails(seller.id, e)}
          label={formatMessage(messages.details)}
        />
        <MenuItem
          icon={<IconShoppingCartSimple />}
          onClick={(e: OnClickEvent) => goToOrders(e, seller.name)}
          label={formatMessage(messages.orders)}
        />
      </Menu>
    </Box>
  )
}

const MoreRender = ({ item, context }: ResolverRenderProps<null, Seller>) => {
  if (context === 'loading') {
    return <Skeleton csx={{ height: 20 }} />
  }

  return <MoreMenu seller={item} />
}
```

[Projeto atualmente em produção](https://sandboxintegracao.myvtex.com/admin/sellers) (as ultimas subidas de versão não estão no ar ainda mas da para vocês verem o uso)
[Projeto no git](https://github.com/vtex/admin-sellers/tree/next) (na branch com as subidas de versão)

**Reply by matheusps** [reply]
Porque tu precisa do  `e` nas suas funções `goToOrders` e `gotToSellerDetails`? O que essas funções fazem?

**Reply by lafray** [reply]
Redireciona para outras paginas.

```

export function goTo(config: {
  event?: OnClickEvent
  href: string
  blank?: boolean
}) {
  const { event, href, blank = false } = config
  const currentWindow = window.top ?? window

  if (blank || event?.ctrlKey || event?.metaKey) {
    currentWindow.open(href, '_blank')
  } else {
    currentWindow.location.href = href
  }
}

export function goToOrders(event: OnClickEvent, sellerName: string) {
  // default period: today
  goTo({
    event,
    href: `/admin/checkout/#/orders?f_sellerNames=${encodeURI(sellerName)}`,
    blank: true,
  })
}

export function goToSellerDetails(sellerId: string, event?: OnClickEvent) {
  localStorage.setItem('previousUrl', getCurrentUrl())
  goTo({ event, href: `/admin/sellers/${encodeURI(sellerId)}` })
}
```

**Reply by matheusps** [reply]
O MenuItem também pode ser um link, isso substitui a necessidade das suas funções. https://codesandbox.io/s/menuitem-as-link-ds9yi7. Um link é melhor por acessibilidade.

**Reply by lafray** [reply]
Legal. Mas da para fazer isso quando uso o render menu para tabela?

**Reply by matheusps** [reply]
Acredito que não! A alternativa seria criar esta opção no menu resolver, tipo um `asLink`. Ao invés do `onClick`, receberia um `href`.

**Reply by lafray** [reply]
E como você enxergam a viabilidade de fazer isso?

**Reply by lafray** [reply]
Acabamos de ter um cenário mais complex no qual a função de goToOrders vai ter que fazer o set de um local storage antes de abrir em um novo link para ir com a pagina filtrada, então já não conseguiria usar o menu com o asLink. Acho que de fato seria importante vocês passarem o evento do onClick no resolver `menu` da tabela

**Reply by lafray** [reply]
Acabamos de ter um cenário mais complex no qual a função de goToOrders vai ter que fazer o set de um local storage antes de abrir em um novo link para ir com a pagina filtrada, então já não conseguiria usar o menu com o asLink. Acho que de fato seria importante vocês passarem o evento do onClick no resolver `menu` da tabela

**Reply by lafray** [reply]
@matheusps

**Comment by davicostalf** [comment]
@lafray Esse problema continua?

**Reply by lafray** [reply]
Foi resolvido
